src/main/java/make_post/ui/MakePostView.java

```diff
@@ -0,0 +1,110 @@
+package make_post.ui;
+
+import make_post.interface_adapters.MakePostController;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MakePostView extends JPanel implements ActionListener, PropertyChangeListener {
+    public static String[] TAGS;
+    private final MakePostController makePostController;
+    private final JList<String> tagsList;
+    private final JTextField enterTitle;
+    private final JTextField enterCollaborators;
+    private final JTextField enterMainDescription;
+    private final JTextField enterDeadline;
+
+    /**
+     * This provides the view for making a Post. It has TextFields for entering all the information necessary for making
+     * a Post.
+     * @param presetTags These are the tags that will be used to filter the Posts.
+     * @param makePostController The controller for the Make Post use case.
+     */
+    public MakePostView(String[] presetTags, MakePostController makePostController){
+        this.TAGS = new String[]{"Sports", "Tech", "Startups"};
```

Comment:
Should this be assigning `presetTags` to `TAGS` instead? Also, I think `TAGS` should be `private static` instead and just add a getter for it if we need to retrieve it.

Reply:
Yes, it should be assigning presetTags (which we instantiate in Main, I think) but since we do not have Main yet, I just used this to check if my code is working. Thanks for flagging this, I completely forgot about it. If I set this to presetTags, then we don't need TAGS anymore.